src/app/core/guards/auth-guard.guard.ts

```diff
@@ -1,25 +1,33 @@
 import { Injectable } from '@angular/core';
 import { CanActivate, Router } from '@angular/router';
 import { AuthService } from '../services/auth/auth.service';
+import { Observable } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthGuard implements CanActivate {
+  private isUserAuthenticated: boolean;
 
   constructor(
-    public auth: AuthService,
+    public authService: AuthService,
     public router: Router
-  ) {}
-
-  public canActivate(): boolean {
-    if (!this.auth.isAuthenticated) {
-      this.router.navigate(['login']);
-
-      return false;
-    }
+  ) {
+    this.authService.isAuthenticated$.subscribe(isAuthenticated => 
+      this.isUserAuthenticated = isAuthenticated
+    );
+  }
 
-    return true;
+  public canActivate(): Observable<boolean> {
+    return new Observable(observer => {
+        if (!this.isUserAuthenticated) {
```

Comment:
will it works if will rewrite?
if (!this.isUserAuthenticated) {
    this.router.navigate(['login']);
}
observer.next(this.isUserAuthenticated)